### PR TITLE
handle bitstypes and wrapped arrays in isleaf

### PR DESCRIPTION
**Pull request by CarloLucibello**
Porting over some desirable behavior from Flux.
Make `MLDataDevices.isleaf(x::AbstractArray{T}) = true` when `T` is a bitstype. Also, make sure wrapped arrays are handled correctly as discussed in https://github.com/FluxML/Flux.jl/issues/2432 and https://github.com/FluxML/Flux.jl/pull/2436

### CODE REVIEW

**Comment by CarloLucibello** [comment]
ready to go

File: Project.toml [hunk 1]
```diff
@@ -53,6 +54,7 @@ Compat = "4.15"
 FillArrays = "1"
 Functors = "0.4.8"
 GPUArrays = "10, 11"
+LinearAlgebra = "1"
```
Comment (avik-pal) [inline Project.toml]:
```suggestion
LinearAlgebra = "1.10"
```

**Comment by CarloLucibello** [comment]
merge?

**Comment by avik-pal** [comment]
Cuda GPU CIs are stalled. Waiting for that to be unblocked

**Pull request merged**